exchange-deployer/src/fee.rs

```diff
@@ -0,0 +1,98 @@
+use crate::action_type::DeployActionType;
+
+multiversx_sc::imports!();
+
+#[multiversx_sc::module]
+pub trait FeeModule: multiversx_sc_modules::pause::PauseModule {
+    #[only_owner]
+    #[endpoint(setActionFee)]
+    fn set_action_fee(
+        &self,
+        type_amount_pairs: MultiValueEncoded<MultiValue2<DeployActionType, BigUint>>,
+    ) {
+        self.require_paused();
+
+        for pair in type_amount_pairs {
+            let (action_type, fee_amount) = pair.into_tuple();
+            require!(action_type != DeployActionType::None, "Invalid action type");
+            self.require_non_zero_fee(&fee_amount);
+
+            self.custom_action_fee(action_type).set(fee_amount);
+        }
+    }
+
+    #[only_owner]
+    #[endpoint(setDefaultActionFee)]
+    fn set_default_action_fee(&self, default_action_fee: BigUint) {
+        self.require_paused();
+        self.require_non_zero_fee(&default_action_fee);
+
+        self.default_action_fee().set(default_action_fee);
+    }
+
+    #[only_owner]
+    #[endpoint(claimFees)]
+    fn claim_fees(&self) {
+        let total_fees = self.total_fees().take();
+        if total_fees == 0 {
+            return;
+        }
+
+        let caller = self.blockchain().get_caller();
+        let fees_token_id = self.fee_token().get();
+        self.send()
+            .direct_esdt(&caller, &fees_token_id, 0, &total_fees);
+    }
+
+    #[view(getActionFee)]
+    fn get_action_fee(&self, action_type: DeployActionType) -> BigUint {
+        let custom_fee = self.custom_action_fee(action_type).get();
+        if custom_fee > 0 {
```

Comment:
Was this a request? This means that you cannot have 0 fee, even if you wanted to.

Reply:
I think so.